webpack.config.js

```diff
@@ -1,34 +1,81 @@
 var ExtractTextPlugin = require('extract-text-webpack-plugin'),
-  path = require('path');
+  webpack = require('webpack'),
+  path = require('path'),
+  nodeEnv = process.env.NODE_ENV || 'production';
 
+
+// TODO: Add in test function: `"test": "NODE_ENV=testing webpack test.js -d --target node && node --require source-map-support/register test-bundle.js #; npm run lint",`
 module.exports = {
-  entry: './src/index.js',
+  entry: nodeEnv === 'testing' ? './test.js' : './src/index.js',
   output: {
-    filename: 'clay-space-edit.js',
-    path: './dist'
+    filename: nodeEnv === 'testing' ? 'test-bundle.js' : 'clay-space-edit.js',
+    path: path.resolve(__dirname, nodeEnv === 'testing' ? '.' : './dist'),
+  },
+  // sub in empty modules for Node built-ins
+  // so webpack doesn't complain about not being able to find the modules
+  // https://github.com/pugjs/pug-loader/issues/8
+  node: {
+    fs: 'empty'
   },
   module: {
-    loaders: [{
-      test: /\.js$/,
-      exclude: /node_modules/,
-      loader: 'babel',
-      query: {
-        presets: ['es2015'],
+    rules: [
+      {
+        test: /\.vue$/,
+        use: [
+          {
+            loader: 'vue-loader',
+            options: {
+              extractCSS: process.env.NODE_ENV === 'production',
+              preserveWhitespace: !process.env.NODE_ENV === 'production',
+              loaders: {
+                css: 'vue-style-loader!css-loader!postcss-loader',
+                sass: 'vue-style-loader!css-loader!postcss-loader!sass-loader'
+              }
+            }
+          }
+        ]
+      },
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        use: [
+          {
+            loader: 'babel-loader',
+            options: {
+              presets: ['es2015']
+            }
+          }
+        ]
+      },
+      {
+        test: /\.scss$/,
+        // don't inject styles when we don't have a window
+        use: nodeEnv === 'testing'
+            ? 'null-loader'
+            : ExtractTextPlugin.extract({
+              fallback: 'style-loader', // backup loader when not building .css file
+              use: [
+                'css-loader',
+                'sass-loader'
+              ]
+            })
+      },
+      {
+        test: /\.svg$/,
+        use: 'raw-loader'
       }
-    }, {
-      test: /\.scss$/,
-      loader: ExtractTextPlugin.extract(
-        'style', // backup loader when not building .css file
-        'css!sass' // loaders to preprocess CSS
-      )
-    }]
+    ]
   },
   resolve: {
     alias: {
       references: path.resolve('./src/services/references.js')
     }
   },
   plugins: [
-    new ExtractTextPlugin('clay-space-edit.css')
-  ]
+    new webpack.EnvironmentPlugin(['NODE_ENV'])
+  ].concat(
+    nodeEnv === 'testing'
```

Comment:
This check is done a few times, might be worth pulling to the top as an `isTestEnv` flag or similar.

Reply:
fixed!